### DISCUSSION TITLE
[Vue] Programmatically controlling style calculations

### DISCUSSION THREAD

**Discussion by diyan-slavov** [floating-ui/floating-ui#2459] [General]
Currently, from the source code, it appears as though the Vue composable for Floating will begin computing the styles when it mounts and they will not be updated after that.

This seems weird to me, because there are many use cases where that will not be the desired outcome - for example, if I'm building a dropdown menu, I may want to defer the calculations until the menu is actually visible, but still have the elements in the DOM.

Currently, an `open` ref can be passed to `useFloating`, and while from the outside it does seem that this is exactly what its meant to do, it only controls the `isPositioned` prop.

Then, there is the `whileElementsMounted` prop, to which I can pass `autoUpdate`, but that doesn't really solve the issue, as it will still recompute the styles when events like resize/scroll are happening, but the popover may not be visible at all.

Is there a reason why there isn't a ref exposed that will control the active state? That way, the  user can tell if its fine to compute it when the element mounts, or pass in his custom ref that will activate it at certain point in time ( for example, on open ).

**Comment by atomiks** [comment]
Conditional rendering (`v-if` instead of `v-show`) is the best choice because then rendering of the DOM elements themselves are deferred until it's visible - the positioning naturally follows this. In the case of `v-show` though, I think you can conditionally set the `ref` attribute of the floating element/set it to nullish value and calculations won't be performed.

**Reply by diyan-slavov** [reply]
Sure, but this is a pretty poor DX. 

Its also common to have Menu/Select elements in the DOM for SEO purposes, so using `v-if` is not always the right choice
